### PR TITLE
2021 01 uniref truncated renderers

### PR DESCRIPTION
**Pull request by aurel-l**
## Purpose
Change the UniRef column renderers for the columns with a truncated list of values

## Approach
Replace ExpandableList with a list + a button linking to the entry itself

## Testing
Updated snapshots

## Checklist
- [ ] My PR is scoped properly, and “does one thing only”
- [x] I have reviewed my own code
- [x] I have checked that linting checks pass and type safety is respected
- [x] I have checked that tests pass and coverage has at least improved, and if not explained the reasons why

Also...
 - Changed the custom eslint rule to handle a edge-case I had forgotten before
 - Created `getEntryPath` and `getEntryPathFor` functions to create URLs more easily for entry pages and applied it in all the column renderers for all namespaces
 - Replaced external links for UniParc XRefs to use Franklin's `ExternalLink` component

### CODE REVIEW

File: src/uniref/config/UniRefColumnConfiguration.tsx [hunk 1]
```diff
@@ -69,42 +65,53 @@ UniRefColumnConfiguration.set(UniRefColumn.commonTaxon, {
...
+        <li key={organism}>{organism}</li>
       ))}
-    </ExpandableList>
+      {organisms.length > CUT_OFF && (
+        <Button
+          element={Link}
+          variant="tertiary"
+          to={getEntryPath(Namespace.uniref, id)}
+        >
+          more organisms
```
Comment (xwatkins) [inline src/uniref/config/UniRefColumnConfiguration.tsx:111]: For consistancy, the "M" of more should be capitalised

File: src/uniref/config/UniRefColumnConfiguration.tsx [hunk 2]
```diff
@@ -137,19 +144,33 @@ UniRefColumnConfiguration.set(UniRefColumn.types, {
...
+        </li>
       ))}
-    </ExpandableList>
+      {members.length > CUT_OFF && (
+        <Button
+          element={Link}
+          variant="tertiary"
+          to={getEntryPath(Namespace.uniref, id)}
+        >
+          {memberCount - CUT_OFF} more member
```
Comment (xwatkins) [inline src/uniref/config/UniRefColumnConfiguration.tsx:169]: For consistancy, the "M" of more should be capitalised

**Review by xwatkins — CHANGES_REQUESTED** [review]
One suggestion, would it be possible to link the "More" links to the "Members" section of the entry directly?

File: src/uniref/config/UniRefColumnConfiguration.tsx (hunk 2 above)
Comment (aurel-l) [reply src/uniref/config/UniRefColumnConfiguration.tsx:169]: Not here since there is a number in front of it

**Comment by aurel-l** [comment]
> One suggestion, would it be possible to link the "More" links to the "Members" section of the entry directly?

I tried, but since the Sequence section is small, the Members section is always in view, and since it's in view, it won't scroll to it. It only scrolls to it if not in view at all.

**Review by xwatkins — APPROVED** [review]
